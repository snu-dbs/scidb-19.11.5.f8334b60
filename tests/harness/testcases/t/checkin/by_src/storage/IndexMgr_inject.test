--setup
--start-query-logging

# Inject Error Support
load_library('misc')

--test
--start-igdata

# inject DI_FAIL_PIN_VALUE
--error --code=scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST \
  "store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,20)), failPinValueArray, _fetch: true)"

# NOTE: Since store() no longer fetches the result array by default
# (SDB-6178), we have to use the undocumented _fetch: option for the
# following tests to pass.  Ideally some operator other than
# store() could be used here.

# inject DI_FAIL_DELETE_RECORD
create temp array failDeleteRecordArray<u:int64,v:int64>[i=0:100:0:10]
store(apply(build(<u:int64>[i=0:100:0:10], i), v, injectError(0,19)), failDeleteRecordArray, _fetch: true)
# Some state from a previous test allocates the injected error object but does not set the flag for the
# injected error.  With that, this test behaves differently when run in isolation versus when run as
# part of the overall injecterror suite.  To work-around, we suppress all output from delete by running
# it in the shell and forcing the return condition to true.  Then, we check for the injected error below.
--shell --command "iquery -c $IQUERY_HOST -p $IQUERY_PORT -aq \
    'delete(failDeleteRecordArray, i!=0)' 2>&1 > /dev/null ; true"
remove(failDeleteRecordArray)
create temp array failDeleteRecordArray<u:int64,v:int64>[i=0:100:0:10]
store(apply(build(<u:int64>[i=0:100:0:10], i), v, injectError(0,19)), failDeleteRecordArray, _fetch: true)
--error --code=scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST \
    "delete(failDeleteRecordArray, i!=0)"
remove(failDeleteRecordArray)

--stop-igdata

--cleanup
--justrun "remove(failPinValueArray)"

--stop-query-logging
