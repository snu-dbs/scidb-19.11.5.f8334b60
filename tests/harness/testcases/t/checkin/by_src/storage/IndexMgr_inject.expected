SCIDB QUERY : <load_library('misc')>
Query was executed successfully

SCIDB QUERY : <store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,20)), failPinValueArray, _fetch: true)>
[An error expected at this place for the query "store(apply(build(<val:int64>[i=0:100:0:10], i), inject, injectError(0,20)), failPinValueArray, _fetch: true)". And it failed with error code = scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST. Expected error code = scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST.]

SCIDB QUERY : <create temp array failDeleteRecordArray<u:int64,v:int64>[i=0:100:0:10]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(apply(build(<u:int64>[i=0:100:0:10], i), v, injectError(0,19)), failDeleteRecordArray, _fetch: true)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <remove(failDeleteRecordArray)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create temp array failDeleteRecordArray<u:int64,v:int64>[i=0:100:0:10]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(apply(build(<u:int64>[i=0:100:0:10], i), v, injectError(0,19)), failDeleteRecordArray, _fetch: true)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <delete(failDeleteRecordArray, i!=0)>
[An error expected at this place for the query "delete(failDeleteRecordArray, i!=0)". And it failed with error code = scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST. Expected error code = scidb::SCIDB_SE_STORAGE::SCIDB_LE_DISK_INDEX_KEY_NOT_EXIST.]

SCIDB QUERY : <remove(failDeleteRecordArray)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <remove(failPinValueArray)>
[SciDB query execution failed. But continuing, as it was intended to just run.]

